tecken/settings.py

```diff
@@ -573,42 +566,95 @@ def filter(self, record):
 ]
 
 
-SYMBOL_URLS = _config(
-    "SYMBOL_URLS",
-    parser=ListOf(str),
-    doc=(
-        "Comma-separated list of urls for symbol downloads.\n\n"
-        "Lookups are performed in list order."
-    ),
-)
-
-UPLOAD_DEFAULT_URL = _config(
-    "UPLOAD_DEFAULT_URL",
-    doc=(
-        "The default url to use for symbol uploads. This must be an item in "
-        "SYMBOL_URLS."
-    ),
-)
-
-UPLOAD_TRY_SYMBOLS_URL = _config(
-    "UPLOAD_TRY_SYMBOLS_URL",
-    doc=(
-        "When an upload comes in with symbols from a Try build, these symbols "
-        "mustn't be uploaded with the regular symbols.\n\n"
-        "You could set this to UPLOAD_DEFAULT_URL with a '/try' prefix.\n\n"
-        "For example::\n\n"
-        "    UPLOAD_DEFAULT_URL=http://s3.example.com/publicbucket/\n"
-        "    UPLOAD_TRY_SYMBOLS_URL=http://s3.example.com/publicbucket/try/"
-    ),
-)
+CLOUD_SERVICE_PROVIDER = _config(
+    "CLOUD_SERVICE_PROVIDER",
+    default="AWS",
+    doc="The cloud service provider Tecken is using. Either AWS or GCP.",
+).upper()
 
-# If UPLOAD_DEFAULT_URL is not in SYMBOL_URLS it's just too weird. It means we'd upload
-# to a S3 bucket we'd never read from and thus it'd be impossible to know the upload
-# worked.
-if UPLOAD_DEFAULT_URL not in SYMBOL_URLS:
-    raise ValueError(
-        f"The UPLOAD_DEFAULT_URL ({UPLOAD_DEFAULT_URL!r}) has to be one of the URLs "
-        f"in SYMBOL_URLS ({SYMBOL_URLS!r})"
+if CLOUD_SERVICE_PROVIDER == "GCS":
+    UPLOAD_GCS_BUCKET = _config(
+        "UPLOAD_GCS_BUCKET",
+        doc="The GCS bucket name for uploads and downloads.",
+    )
+    DOWNLOAD_S3_BUCKET = _config(
+        "DOWNLOAD_S3_BUCKET",
+        raise_error=False,
+        doc="The S3 bucket name for downloads.",
+    )
+    DOWNLOAD_S3_REGION = (
+        _config(
+            "DOWNLOAD_S3_REGION",
+            raise_error=False,
+            doc="The region of the S3 download bucket.",
+        )
+        or None
```

Comment:
For my notes: We want this to default to `None` which we can do this way (converts the `NO_VALUE` to a `None` via the `or`) or using an Everett configuration parser. We chose the `or None`. We need the default to be `None` because that's what we need to pass as a value for the `region` argument when creating an s3 client.